### PR TITLE
adding methods in VMenuTree with Exposed & create DBSchema

### PR DESCRIPTION
**Pull request by h-haddad**

### CODE REVIEW

File: galite-core/src/main/kotlin/org/kopi/galite/visual/VMenuTree.kt [hunk 1]
```diff
@@ -332,7 +459,25 @@ class VMenuTree(ctxt: DBContext,
    * Fetches the favorites from the database.
    */
   private fun fetchFavorites() {
-    TODO()
+    transaction {
+      val query = if (isSuperUser && menuTreeUser != null) {
+
+        val id = UserRights.select { UserRights.id eq 0 }.single()[UserRights.id]
```
Comment (ZMarouani) [inline galite-core/src/main/kotlin/org/kopi/galite/visual/VMenuTree.kt:465]: Should this be in this if {} ?

**Review by ZMarouani — CHANGES_REQUESTED** [review]

File: galite-core/src/main/kotlin/org/kopi/galite/visual/VMenuTree.kt [hunk 2]
```diff
@@ -300,23 +312,138 @@ class VMenuTree(ctxt: DBContext,
...
-    TODO()
+    val localModules: ArrayList<Module> = ArrayList()
+    var icon: String? = null
+
+    transaction {
+      SELECT_MODULES.forEach {
+        if (it[Modules.symbol] != null && it[Modules.symbol] != 0) {
+          val symbol = it[Modules.symbol] as Int
+
+          icon = Symbols.select { Symbols.id eq symbol }.single()[Symbols.objectName]
```
[the remark this answers is not in the document]
Comment (h-haddad) [reply galite-core/src/main/kotlin/org/kopi/galite/visual/VMenuTree.kt:323]: I do not develop this class just I added exposed

File: galite-core/src/main/kotlin/org/kopi/galite/visual/VMenuTree.kt [hunk 3]
```diff
@@ -300,23 +312,138 @@ class VMenuTree(ctxt: DBContext,
...
+        }
+
+        val module = Module(it[Modules.id],
+                it[Modules.father],
+                it[Modules.shortName],
+                it[Modules.sourceName],
+                it[Modules.objectName],
```
[the remark this answers is not in the document]
Comment (h-haddad) [reply galite-core/src/main/kotlin/org/kopi/galite/visual/VMenuTree.kt:330]: No this method is used in Query class

File: galite-core/src/main/kotlin/org/kopi/galite/visual/VMenuTree.kt [hunk 4]
```diff
@@ -50,10 +61,11 @@ class VMenuTree(ctxt: DBContext,
                 loadFavorites: Boolean = true) : VWindow(ctxt) {
```
[the remark this answers is not in the document]
Comment (h-haddad) [reply galite-core/src/main/kotlin/org/kopi/galite/visual/VMenuTree.kt:62]: I do not develop this class just I added exposed

**Review by ZMarouani — APPROVED** [review]

**Pull request merged**